warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java

```diff
@@ -2477,8 +2491,304 @@ public Equality getEquality() {
         return equality;
     }
 
-    public boolean isSeekingEventAggregation() {
-        return seekingEventAggregation;
+    private static final Joiner COMMA_JOINER = Joiner.on(",").skipNulls();
+
+    // Map of class strings to DefaultOptions implementations.
+    protected static final Map<String,DefaultOptions> defaultOptionsMap = new HashMap<>();
+
+    static {
+        // Add an initial default options instance for QueryOptions.
+        defaultOptionsMap.put(QueryOptions.class.getName(), new QueryOptions().createDefaultOptions());
+    }
+
+    /**
+     * Used to populate the defaultOptionsMap with the default options for each class. When overriding this method, call super.createDefaultOptions() to ensure
+     * the default options are populated. Changes to these defaults will be reflected in the DefaultOptions object returned by this method. Add them with
+     * defaultOptions.putDefaultValue(key, value).
+     *
+     * @return the default options
+     */
+    protected DefaultOptions createDefaultOptions() {
```

Comment:
Can we create a test to verify that this set of options is complete?  See the ShardQueryConfiguratTest on how that might be done using the ObjectMapper.